baseplate/frameworks/pyramid/__init__.py

```diff
@@ -189,8 +190,9 @@ def _on_new_request(self, event: pyramid.events.ContextFound) -> None:
         if self.header_trust_handler.should_trust_edge_context_payload(request):
             try:
                 edge_payload = request.headers.get("X-Edge-Request", None)
-                if self.edge_context_factory:
-                    edge_context = self.edge_context_factory.from_upstream(edge_payload)
+                if self.edge_context_factory and edge_payload:
+                    edge_payload_decoded = base64.b64decode(edge_payload)
```

Comment:
I think this is going to be a `string` and `b64decode` wants `bytes`, so you'll want to `.encode()` edge payload here.